### PR TITLE
Practice JS CSS HTML

### PR DESCRIPTION
**Pull request by DmytroP888**
Sorting on the page. Accordion.

### CODE REVIEW

File: script download new array_1.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+//                          <div> <h4> Тип: ${products[i].type}. Категорія: ${products[i].category}. 
+//                          Сезон: ${products[i].season}. Стать: ${products[i].sex}. </h4> ${descp} </div>
+//                          </li></a>`;
+//             document.getElementById("lot1").innerHTML += stroka;
+//         }
+//     }
+// }
+
+// ---------------------  ACCORDION  ---------------------
+// -------  Sorting block mune accordion (works with method 1)  --------
```
Comment (andriiivaniuk) [inline script download new array_1.js:89]: why so many commented out code?

File: script download new array_1.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    for (let i = 0; i < products.length; i++) {
+        if (vvod !== '') {
+            const poda = document.querySelectorAll(".productPlus");
+            poda[i].style.display = 'none';
+        }
+        if (vvod !== '' &&
+            vvod == products[i].type ||
+            vvod == products[i].category ||
+            vvod == products[i].season ||
+            vvod == products[i].sex) {
```
Comment (andriiivaniuk) [inline script download new array_1.js:34]: it's safer to use === to make sure that comparable data is the same

File: script download new array_1.js (hunk 1 above)
Comment (DmytroP888) [reply script download new array_1.js:89]: The commented code is another sorting method written for the purpose of learning.

File: script download new array_1.js (hunk 2 above)
Comment (DmytroP888) [reply script download new array_1.js:34]: Yes really! === So it will be better!))

File: Sort prod download new array_1.html [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+</head>
+
+<body>
+    <div class="menu">
+        <div class="menuButton">
+            <input type="text" name="name by sort" id="inp" class="pole"></input>
+            <br>
+            <button class="sc" id="btnSelect" onclick="sort()">Select</button>
+            <button class="sc" id="btnClear" onclick="clearon()">Сlear all</button>
+            <button class="sc" id="allProds">All prod.s</button>
```
Comment (vladhybalo) [inline Sort prod download new array_1.html:21]: class `cs` doesn't look very informative, could you pls change it to more human-readable?

File: Sort prod download new array_1.html [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                    <select name="season" id="">
+                        <option name="season" id="formSeasonidlito" value="літо">літо</option>
+                        <option name="season" id="formSeasonidosin" value="осінь">осінь</option>
+                        <option name="season" id="formSeasonidzuma" value="зима">зима</option>
+                        <option name="season" id="formSeasonidvesna" value="весна">весна</option>
+                    </select>
+                </p>
+                <p class="inputModal"><label>Стать: </label>
+                    <select name="sex" id="">
+                        <option name="sex" id="formSexjinka" value="жінка">жінка</option>
```
Comment (vladhybalo) [inline Sort prod download new array_1.html:90]: Pls dont use transliteration `jinka` and so on

File: script download new array_1.js [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+}
+
+// ================== Api array =================
+let products = null;
+const apiArray = async() => {
+    try {
+        const response = await fetch('http://localhost:3000/products');
+        if (response.ok) {
+            const ggg = await response.json();
+            products = ggg;
```
Comment (vladhybalo) [inline script download new array_1.js:14]: ggg redundant)

File: script download new array_1.js [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+            console.log("~ result", "Список успешно загружен")
+        }
+    } catch (error) {
+        console.log("~ error", 'Список не был загружен!!!' + error);
+    }
+}
+
+// ============================ OUTPUT allarray start ====================
+function allarray() {
```
Comment (vladhybalo) [inline script download new array_1.js:23]: pls use cameCase

File: script download new array_1.js [hunk 7]
```diff
@@ -0,0 +1,500 @@
...
+document.getElementById("allProds").addEventListener("click", outputAll);
+
+outputAll();
+// ------------------------------------------------------------------------------
+
+let filterType = null;
+
+document.addEventListener('change', badu);
+
+// ================== Grouping function with tags for sorting =================
```
Comment (vladhybalo) [inline script download new array_1.js:59]: why not to name the function in same way you comment it

File: script download new array_1.js [hunk 8]
```diff
@@ -0,0 +1,500 @@
...
+        document.getElementById("lot1").innerHTML += stroka;
+    })
+}
+//====================== ACCORDION menu select out ===========================================
+document.querySelector(".menuProd")
+    .addEventListener('click', event => {
+        if (event.target.className === 'name_Prods') { // Використано правильний метод делегування парента до елемента з подією
+            namesort = event.target.getAttribute('data-name');
+            clearon();
+            filterType = products.filter((e, i) => e.type === namesort || e.category === namesort || e.season === namesort || e.sex === namesort);
```
Comment (vladhybalo) [inline script download new array_1.js:172]: there is `find` function

File: script download new array_1.js (hunk 6 above)
Comment (vladhybalo) [reply script download new array_1.js:23]: `allarray` -> `renderAllProducts`

File: script download new array_1.js [hunk 9]
```diff
@@ -0,0 +1,500 @@
...
+                          Сезон: ${products[i].season}. Стать: ${products[i].sex}.
+                          </h4> ${products[i].description} </div></li>`;
+        document.getElementById("lot1").innerHTML += stroka;
+    })
+}
+
+function outputAll() {
+    clearon();
+    if (products === null) {
+        apiArray()
```
Comment (vladhybalo) [inline script download new array_1.js:36]: `apiArray` -> `fetchProducts`

File: script download new array_1.js [hunk 10]
```diff
@@ -0,0 +1,500 @@
+// ------------  Function to clear all products from the page  -----------
+clearon = () => {
```
Comment (vladhybalo) [inline script download new array_1.js:2]: add 'use strict' at the top of the file and you will see that creating variables without special work like `let, const` is deprecated

File: script download new array_1.js [hunk 11]
```diff
@@ -0,0 +1,500 @@
...
+            lonik();
+        }
+    });
+
+//==================== Function select sort  ===================
+function badu(event) {
+    if (event.target.className === "selectSort") {
+        let targetSelect = event.target.value;
+        const confirmSelect = alert('Ви обрали для сортування: ' + event.target.value);
+        newfilterType = filterType.filter((e, i) => e.type === targetSelect || e.category === targetSelect || e.season === targetSelect || e.sex === targetSelect);
```
Comment (vladhybalo) [inline script download new array_1.js:182]: newfilterType is redundant

File: script download new array_1.js [hunk 12]
```diff
@@ -0,0 +1,500 @@
...
+    filterType.forEach((e, i) => {
+
+        // -------- Type select ---------
+        const sortTypeSelekt = arrayTypTransform.filter(e => e !== filterType[i].type);
+        let liteType = '';
+        sortTypeSelekt.forEach((element, i) => {
+            liteType += '<option value="' + sortTypeSelekt[i] + '">' + sortTypeSelekt[i] + '</option>';
+        })
+        const litlTypeOne = `<option> ${filterType[i].type} </option>`;
+        let typeSelect = '<label>Тип: </label><select class="selectSort" id="idType">' + litlTypeOne + liteType + '</select>';
```
Comment (vladhybalo) [inline script download new array_1.js:99]: If code logic is the same as below try to make a function for this

File: script download new array_1.js (hunk 10 above)
Comment (vladhybalo) [reply script download new array_1.js:2]: clearon -> productsUICleanup or productsPageCleanup

File: script download new array_1.js [hunk 13]
```diff
@@ -0,0 +1,500 @@
...
+        if (typeTwo < 3) {
+            typeSelect = 'Тип: ' + filterType[i].type + '.' + '  ';
+        }
+        if (categoryTwo < 3) {
+            categorySelect = 'Категорія: ' + filterType[i].category + '.' + '  ';
+        }
+        if (seasonTwo < 3) {
+            seasonSelect = 'Сезон: ' + filterType[i].season + '.' + '  ';
+        }
+        if (sexTwo < 3) {
```
Comment (vladhybalo) [inline script download new array_1.js:153]: cant get why do you need this checks

File: script download new array_1.js [hunk 14]
```diff
@@ -0,0 +1,500 @@
...
+        const confirmSelect = alert('Ви обрали для сортування: ' + event.target.value);
+        newfilterType = filterType.filter((e, i) => e.type === targetSelect || e.category === targetSelect || e.season === targetSelect || e.sex === targetSelect);
+        filterType = newfilterType;
+        clearon()
+        lonik()
+    }
+}
+
+//============= Grouping function via input field  =============
+function sort() {
```
Comment (vladhybalo) [inline script download new array_1.js:190]: its not sort at all, its rather filteringProducts

File: script download new array_1.js [hunk 15]
```diff
@@ -0,0 +1,500 @@
...
+    let oood = document.getElementById("inp").value;
```
Comment (vladhybalo) [inline script download new array_1.js:191]: oood -> filterInput

File: script download new array_1.js [hunk 16]
```diff
@@ -0,0 +1,500 @@
...
+}
+
+// ================  Menu accordion CSS  ===================
+document.querySelectorAll(".accordion")
+    .forEach((e) => {
+        e.addEventListener('click', () => {
+            // accordion.forEach(el => {el.classList.remove('show')});
+            e.classList.toggle('show');
+        });
+    });
```
Comment (vladhybalo) [inline script download new array_1.js:208]: pls use delegation, there shouldnt be so many handlers, just one on the parent

File: script download new array_1.js [hunk 17]
```diff
@@ -0,0 +1,500 @@
...
+function openFormModal() {
+    document.getElementById('boxModal').classList.toggle('open');
+}
+
+// --------------------- Form products property ------------------------
+const form = document.getElementById("myForm");
+
+form.addEventListener('submit', (event) => {
+    event.preventDefault();
+    var elements = document.getElementById("myForm").elements;
```
Comment (vladhybalo) [inline script download new array_1.js:222]: image

File: script download new array_1.js [hunk 18]
```diff
@@ -0,0 +1,500 @@
...
+    var obj = {};
+    for (var i = 0; i < elements.length; i++) {
+        if (elements.item(i).className !== 'btnModal') {
```
Comment (vladhybalo) [inline script download new array_1.js:225]: I would rather get all needed elements by more specific selector

File: script download new array_1.js [hunk 19]
```diff
@@ -0,0 +1,500 @@
...
+
+
+
+// ================= Valid form  =========================
+const btnSubmit = document.getElementById('btnModalok');
+const formMost = document.getElementById('myForm');
+
+
+// _____________________________ Input Category FORM ___________________________
+function formCatgInpReg() {
```
Comment (vladhybalo) [inline script download new array_1.js:249]: validateCategoryInput

File: script download new array_1.js [hunk 20]
```diff
@@ -0,0 +1,500 @@
...
+    } else
+    if (fImgRegB.length >= 1 && fImgRegB.length < 3) {
+        inputImgB.setCustomValidity("Enter at least < 3 characters");
+        inputImgB.reportValidity();
+        return false;
+    } else if (fImgRegB.length >= 3 && !fImgRegB.match(fImgRegularB)) {
+        inputImgB.setCustomValidity("Incorrect input!");
+        document.getElementById('formImgid').classList.add('formInputRed');
+        return false;
+    }
```
Comment (vladhybalo) [inline script download new array_1.js:322]: looks like you has very common logic for validation, try to some base function with common logic

File: script download new array_1.js [hunk 21]
```diff
@@ -0,0 +1,500 @@
...
+    event.preventDefault();
+    var elements = document.getElementById("myForm").elements;
+    var obj = {};
+    for (var i = 0; i < elements.length; i++) {
+        if (elements.item(i).className !== 'btnModal') {
+            var item = elements.item(i);
+            obj[item.name] = item.value;
+        }
+    }
+    let poroh = JSON.stringify(obj)
```
Comment (vladhybalo) [inline script download new array_1.js:230]: poroh -> product

File: script download new array_1.js [hunk 22]
```diff
@@ -0,0 +1,500 @@
...
+        allarray();
+    }
+}
+
+document.getElementById("allProds").addEventListener("click", outputAll);
+
+outputAll();
+// ------------------------------------------------------------------------------
+
+let filterType = null;
```
Comment (vladhybalo) [inline script download new array_1.js:55]: filterType -> filteredProducts

**Pull request merged**